src/contracts/middleware/BLSSignatureChecker.sol

```diff
@@ -1,508 +1,218 @@
-        assembly {
-            success := staticcall(sub(gas(), 2000), 7, add(input, 0x40), 0x60, add(input, 0x80), 0x40)
-        }
-        require(success, "BLSSignatureChecker.checkSignatures: generator random shift failed");
-
-        // (input[6], input[7]) = (input[4], input[5]) + (input[6], input[7]) = g1 * gamma + H(m)
-        // solium-disable-next-line security/no-inline-assembly
-        assembly {
-            success := staticcall(sub(gas(), 2000), 6, add(input, 0x80), 0x80, add(input, 0xC0), 0x40)
-        }
-        require(success, "BLSSignatureChecker.checkSignatures: generator random shift and G1 hash addition failed");
-
-        // insert negated coordinates of the generator for G2
-        input[2] = BN254.nG2x1;
-        input[3] = BN254.nG2x0;
-        input[4] = BN254.nG2y1;
-        input[5] = BN254.nG2y0;
-
-        // in summary
-        // (input[0], input[1]) =  sigma + gamma * signingPublicKey
-        // (input[2], input[3], input[4], input[5]) = negated generator of G2
-        // (input[6], input[7]) = g1 * gamma + H(m)
-        // (input[8], input[9], input[10], input[11]) = public key in G2
-        
-        
-        /**
-         * @notice now we verify that e(sigma + gamma * pk, -g2)e(H(m) + gamma * g1, pkG2) == 1
-         */
-
-        assembly {
-            // check the pairing; if incorrect, revert                
-            // staticcall address 8 (ecPairing precompile), forward all gas, send 384 bytes (0x180 in hex) = 12 (32-byte) inputs.
-            // store the return data in input[0], and copy only 32 bytes of return data (since precompile returns boolean)
-            success := staticcall(sub(gas(), 2000), 8, input, 0x180, input, 0x20)
-        }
-        require(success, "BLSSignatureChecker.checkSignatures: pairing precompile call failed");
-        // check that the provided signature is correct
-        require(input[0] == 1, "BLSSignatureChecker.checkSignatures: Pairing unsuccessful");
-
-        emit SignatoryRecord(
-            msgHash,
-            taskNumberToConfirm,
-            signedTotals.signedStakeFirstQuorum,
-            signedTotals.signedStakeSecondQuorum,
-            pubkeyHashes
-        );
-
-        // set compressedSignatoryRecord variable used for fraudproofs
-        compressedSignatoryRecord = MiddlewareUtils.computeSignatoryRecordHash(
-            taskNumberToConfirm,
-            pubkeyHashes,
-            signedTotals.signedStakeFirstQuorum,
-            signedTotals.signedStakeSecondQuorum
+            // verify the signature
+            (bool pairingSuccessful, bool sigantureIsValid) = trySignatureAndApkVerification(msgHash, apk, nonSignerStakesAndSignature.apkG2, nonSignerStakesAndSignature.sigma);
+            require(pairingSuccessful, "BLSSignatureChecker.checkSignatures: pairing precompile call failed");
+            require(sigantureIsValid, "BLSSignatureChecker.checkSignatures: signature is invalid");
```

Comment:
typo: `siganture` instead of `signature`
also this line seems too long and should be broken up

Reply:
resolved https://github.com/Layr-Labs/eigenlayer-contracts/pull/35/commits/7dec57d13c4154092cffb3c92bebf98e0e0199ad